### PR TITLE
refactor Permission classmethods

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Refactor permission generation. The existing way of 0b integers is prone to breaking and is hard to read.
Additionally, adds a missing events classmethod.
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by VincentRPS** [comment]
what does this do exactly?

**Comment by onerandomusername** [comment]
> what does this do exactly?

Overall, makes the code easier to read, as it can now be seen exactly what permissions are being set. The previous style was manually created binary, which is prone to error and needs to be updated when a new permission is added.

**Comment by VincentRPS** [comment]
> > what does this do exactly?
> 
> Overall, makes the code easier to read, as it can now be seen exactly what permissions are being set. The previous style was manually created binary, which is prone to error and needs to be updated when a new permission is added.

how is it prone to error?

**Comment by Victorsitou** [comment]
well, you might type an extra one, and you won't notice

**Review by EQUENOS — APPROVED** [review]

File: disnake/permissions.py [hunk 1]
```diff
@@ -238,44 +286,94 @@ def text(cls: Type[P]) -> P:
...
+    @cached_creation
     def stage_moderator(cls: Type[P]) -> P:
         """A factory method that creates a :class:`Permissions` with all
         "Stage Moderator" permissions from the official Discord UI set to ``True``.
 
         .. versionadded:: 1.7
         """
-        return cls(0b100000001010000000000000000000000)
+        return cls(
+            manage_channels=True,
```
Comment (shiftinv) [inline disnake/permissions.py:349]:
```suggestion
            request_to_speak=True,
```

File: disnake/permissions.py [hunk 2]
```diff
@@ -198,9 +213,24 @@ def all_channel(cls: Type[P]) -> P:
         .. versionchanged:: 2.3
             Added :attr:`start_embedded_activities` permission.
         """
-        return cls(0b1111110110110011111101111111111101010001)
+        guild_specific_perms = {
+            "administrator",
+            "ban_members",
+            "change_nickname",
+            "kick_members",
+            "manage_emojis",
```
Comment (shiftinv) [inline disnake/permissions.py:221]:
```suggestion
            "manage_emojis",
            "manage_events",
```
(also missing from the docstring)

File: disnake/permissions.py [hunk 3]
```diff
@@ -198,9 +213,24 @@ def all_channel(cls: Type[P]) -> P:
...
+            "manage_guild",
+            "manage_nicknames",
+            "moderate_members",
+            "view_audit_log",
+            "view_guild_insights",
+        }
+        instance = cls.all()
+        instance.update(**dict.fromkeys(guild_specific_perms, False))  # type: ignore
```
Comment (shiftinv) [inline disnake/permissions.py:229]:
```suggestion
        instance.update(**dict.fromkeys(guild_specific_perms, False))
```
doesn't seem to be required

**Review by shiftinv — CHANGES_REQUESTED** [review]

File: disnake/permissions.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/permissions.py:221]: manage events is a per-channel permission.

File: disnake/permissions.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/permissions.py]: https://github.com/DisnakeDev/disnake/pull/286/commits/6cc1ab5c497b37df099f6f3dc209764798e55935

**Review by shiftinv — APPROVED** [review]

**Pull request merged**